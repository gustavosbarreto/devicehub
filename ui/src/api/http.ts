import axios, { AxiosInstance } from "axios";
import * as axiosTs from "./client";
import { Configuration } from "./client";
import { BaseAPI } from "./client/base";
import { setupInterceptorsTo } from "./interceptors";

// This is the default configuration for local instance endpoints
const configuration = new Configuration();
configuration.basePath = `${window.location.protocol}//${window.location.host}`;
configuration.accessToken = localStorage.getItem("token") || "";

// We need a custom configuration for cloud endpoints
const cloudApiConfiguration = new Configuration();
cloudApiConfiguration.basePath = "https://cloud.shellhub.io";

// Creates a new axios instance and setup interceptors by default
const newAxiosInstance = (setupInterceptor = true): AxiosInstance => {
  const instance = axios.create();
  if (setupInterceptor) setupInterceptorsTo(instance);
  return instance;
};
const sessionsApi = new axiosTs.SessionsApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const devicesApi = new axiosTs.DevicesApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const defaultApi = new axiosTs.DefaultApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const namespacesApi = new axiosTs.NamespacesApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const apiKeysApi = new axiosTs.ApiKeysApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const sshApi = new axiosTs.SshApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const tagsApi = new axiosTs.TagsApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const usersApi = new axiosTs.UsersApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const mfaApi = new axiosTs.MfaApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const billingApi = new axiosTs.BillingApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const rulesApi = new axiosTs.RulesApi(
  configuration,
  undefined,
  newAxiosInstance(),
);
const announcementApi = new axiosTs.AnnouncementsApi(
  cloudApiConfiguration,
  undefined,
  newAxiosInstance(false),
);

/**
 * @deprecated This method is deprecated and no longer performs any action,
 * kept for backward compatibility but it will be removed in the future.
 **/
export const createNewClient = () => Function;

/**
 * Extends the interface BaseAPI to include a new method `getAxios` allowing
 * access to the protected axios property without the need to access it
 * directly from outside the class and avoiding a linting error caused by
 * accessing a protected property.
 **/
declare module "./client/base" {
  interface BaseAPI {
    getAxios(): AxiosInstance;
  }
}

/** Returns the axios instance */
BaseAPI.prototype.getAxios = function getAxios(this: BaseAPI): AxiosInstance {
  return this.axios;
};

export {
  configuration,
  sessionsApi,
  devicesApi,
  defaultApi,
  namespacesApi,
  apiKeysApi,
  sshApi,
  tagsApi,
  usersApi,
  mfaApi,
  billingApi,
  rulesApi,
  announcementApi,
};
