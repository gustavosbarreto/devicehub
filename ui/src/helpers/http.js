import Axios from 'axios'
import store from '../store'
import router from "../router";

export default () => {
    let axios = Axios.create({
        baseURL: `${location.protocol}//${location.host}/api`,
        headers: {
            Authorization: `Bearer ${localStorage.getItem('token')}`
        }
    })

    axios.interceptors.response.use(
        response => response,
        (error) => {
            if (error.response.status === 401) {
                store.dispatch('auth/logout')
                router.push('/login')
            }
        }
    )

    return axios
}
