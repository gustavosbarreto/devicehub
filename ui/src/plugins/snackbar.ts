import { App, Plugin } from "vue";

const InjectionKey = "snackbar";

interface PluginInterface {
  showInfo(message: string);
  showSuccess(message: string);
  showWarning(message: string);
  showError(message: string);
}

const SnackbarPlugin: Plugin = {
  install(app: App) {
    const store = app.config.globalProperties.$store;

    const plugin = {
      showInfo(message: string) {
        store.commit("snackbar/showMessage", { type: "info", message });
      },
      showSuccess(message: string) {
        store.commit("snackbar/showMessage", { type: "success", message });
      },
      showWarning(message: string) {
        store.commit("snackbar/showMessage", { type: "warning", message });
      },
      showError(message: string) {
        store.commit("snackbar/showMessage", { type: "error", message });
      },
    } as PluginInterface;

    app.provide(InjectionKey, plugin);
  },
};

export { InjectionKey, type PluginInterface, SnackbarPlugin };
