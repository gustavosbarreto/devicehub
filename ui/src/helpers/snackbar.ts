import { inject } from "vue";
import { PluginInterface, InjectionKey } from "@/plugins/snackbar";

function useSnackbar(): PluginInterface {
  return inject(InjectionKey) as PluginInterface;
}

export default useSnackbar;
