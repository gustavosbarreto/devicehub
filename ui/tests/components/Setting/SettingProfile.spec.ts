import { createVuetify } from "vuetify";
import { flushPromises, mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";
import MockAdapter from "axios-mock-adapter";
import { nextTick } from "vue";
import SettingProfile from "@/components/Setting/SettingProfile.vue";
import { namespacesApi, usersApi } from "@/api/http";
import { store, key } from "@/store";
import { router } from "@/router";
import { envVariables } from "@/envVariables";
import { SnackbarPlugin } from "@/plugins/snackbar";

type SettingProfileWrapper = VueWrapper<InstanceType<typeof SettingProfile>>;

describe("Settings Namespace", () => {
  let wrapper: SettingProfileWrapper;

  const vuetify = createVuetify();

  let mockNamespace: MockAdapter;

  let mockUser: MockAdapter;

  const members = [
    {
      id: "xxxxxxxx",
      username: "test",
      role: "owner",
    },
  ];

  const namespaceData = {
    name: "test",
    owner: "test",
    tenant_id: "fake-tenant-data",
    members,
    settings: {
      session_record: true,
    },
    max_devices: 3,
    devices_count: 3,
    created_at: "",
  };

  const authData = {
    status: "success",
    token: "",
    user: "test",
    name: "test",
    tenant: "fake-tenant-data",
    email: "test@test.com",
    recovery_email: "test2@test.com",
    id: "xxxxxxxx",
    role: "owner",
    mfa: {
      enable: false,
      validate: false,
    },
  };

  const session = true;

  beforeEach(async () => {
    vi.useFakeTimers();
    localStorage.setItem("tenant", "fake-tenant-data");
    envVariables.isCloud = true;

    mockNamespace = new MockAdapter(namespacesApi.getAxios());
    mockUser = new MockAdapter(usersApi.getAxios());

    mockNamespace.onGet("http://localhost:3000/api/namespaces/fake-tenant-data").reply(200, namespaceData);
    mockUser.onGet("http://localhost:3000/api/users/security").reply(200, session);
    mockUser.onGet("http://localhost:3000/api/auth/user").reply(200, authData);

    store.commit("auth/authSuccess", authData);
    store.commit("auth/changeData", authData);
    store.commit("namespaces/setNamespace", namespaceData);
    store.commit("security/setSecurity", session);

    wrapper = mount(SettingProfile, {
      global: {
        plugins: [[store, key], vuetify, router, SnackbarPlugin],
        config: {
          errorHandler: () => { /* ignore global error handler */ },
        },
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Data is defined", () => {
    expect(wrapper.vm.$data).toBeDefined();
  });

  it("Renders components", async () => {
    expect(wrapper.find('[data-test="account-header"]').exists());
    expect(wrapper.find('[data-test="change-data-btn"]').exists());
    expect(wrapper.find('[data-test="cancel-btn"]').exists());
    expect(wrapper.find('[data-test="update-user-btn"]').exists());
    expect(wrapper.find('[data-test="name-text"]').exists());
    expect(wrapper.find('[data-test="username-text"]').exists());
    expect(wrapper.find('[data-test="email-text"]').exists());
    expect(wrapper.find('[data-test="recovery-email-text"]').exists());
    expect(wrapper.find('[data-test="password-header"]').exists());
    expect(wrapper.find('[data-test="change-password-btn"]').exists());
    expect(wrapper.find('[data-test="cancel-password-btn"]').exists());
    expect(wrapper.find('[data-test="update-password-btn"]').exists());
    expect(wrapper.find('[data-test="password-text"]').exists());
    expect(wrapper.find('[data-test="newPassword-text"]').exists());
    expect(wrapper.find('[data-test="confirmNewPassword-text"]').exists());
  });

  it("Successfully changes user data", async () => {
    const changeUserData = {
      recovery_email: "test2@test.com",
      email: "test@test.com",
      name: "test",
      username: "test",
    };

    mockUser.onPatch("http://localhost:3000/api/users").reply(200);

    await wrapper.findComponent('[data-test="change-data-btn"]').trigger("click");
    await wrapper.findComponent('[data-test="name-text"]').setValue("test");
    await wrapper.findComponent('[data-test="username-text"]').setValue("test");
    await wrapper.findComponent('[data-test="email-text"]').setValue("test@test.com");
    await wrapper.findComponent('[data-test="recovery-email-text"]').setValue("test2@test.com");

    const changeDataSpy = vi.spyOn(store, "dispatch");
    await wrapper.findComponent('[data-test="update-user-btn"]').trigger("click");

    vi.runOnlyPendingTimers();

    await nextTick();
    await flushPromises();
    expect(changeDataSpy).toHaveBeenCalledWith("users/patchData", changeUserData);
  });

  it("Successfully changes user password", async () => {
    const changePasswordData = {
      recovery_email: "",
      email: "",
      name: "",
      username: "",
      currentPassword: "test",
      newPassword: "test123",
    };

    await wrapper.findComponent('[data-test="change-password-btn"]').trigger("click");

    mockUser.onPatch("http://localhost:3000/api/users").reply(200);

    const changePasswordSpy = vi.spyOn(store, "dispatch");
    await wrapper.findComponent('[data-test="password-text"]').setValue("test");
    await wrapper.findComponent('[data-test="newPassword-text"]').setValue("test123");
    await wrapper.findComponent('[data-test="confirmNewPassword-text"]').setValue("test123");

    await wrapper.findComponent('[data-test="update-password-btn"]').trigger("click");

    vi.runOnlyPendingTimers();

    await nextTick();
    await flushPromises();
    expect(changePasswordSpy).toHaveBeenCalledWith("users/patchPassword", changePasswordData);
  });

  it("Fails changes user data", async () => {
    const changeUserData = {
      recovery_email: "test2@test.com",
      email: "test@test.com",
      name: "test",
      username: "test",
    };

    mockUser.onPatch("http://localhost:3000/api/users").reply(401);

    const changeDataSpy = vi.spyOn(store, "dispatch");
    await wrapper.findComponent('[data-test="change-data-btn"]').trigger("click");
    await wrapper.findComponent('[data-test="name-text"]').setValue("test");
    await wrapper.findComponent('[data-test="username-text"]').setValue("test");
    await wrapper.findComponent('[data-test="email-text"]').setValue("test@test.com");
    await wrapper.findComponent('[data-test="recovery-email-text"]').setValue("test2@test.com");
    await flushPromises();

    await wrapper.findComponent('[data-test="update-user-btn"]').trigger("click");

    vi.runOnlyPendingTimers();

    await nextTick();
    await flushPromises();
    expect(changeDataSpy).toHaveBeenCalledWith("users/patchData", changeUserData);
  });

  it("Fails changes user password", async () => {
    const changePasswordData = {
      recovery_email: "",
      email: "",
      name: "",
      username: "",
      currentPassword: "test",
      newPassword: "test123",
    };

    await wrapper.findComponent('[data-test="change-password-btn"]').trigger("click");

    mockUser.onPatch("http://localhost:3000/api/users").reply(401);

    const changePasswordSpy = vi.spyOn(store, "dispatch");
    await wrapper.findComponent('[data-test="password-text"]').setValue("test");
    await wrapper.findComponent('[data-test="newPassword-text"]').setValue("test123");
    await wrapper.findComponent('[data-test="confirmNewPassword-text"]').setValue("test123");

    await wrapper.findComponent('[data-test="update-password-btn"]').trigger("click");

    vi.runOnlyPendingTimers();

    await nextTick();
    await flushPromises();
    expect(changePasswordSpy).toHaveBeenCalledWith("users/patchPassword", changePasswordData);
  });
});
