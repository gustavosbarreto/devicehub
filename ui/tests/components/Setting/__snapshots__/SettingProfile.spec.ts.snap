// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Settings Namespace > Renders the component 1`] = `
"<div data-v-b3fa301d=\\"\\" class=\\"v-container v-locale--is-ltr\\">
  <div data-v-b3fa301d=\\"\\" class=\\"v-row align-center justify-center\\">
    <div data-v-b3fa301d=\\"\\" class=\\"v-col-sm-8 v-col\\">
      <div data-v-b3fa301d=\\"\\" class=\\"v-row\\">
        <div data-v-b3fa301d=\\"\\" class=\\"v-col\\" data-test=\\"account-header\\">
          <h3 data-v-b3fa301d=\\"\\">Account</h3>
        </div>
        <div data-v-b3fa301d=\\"\\" class=\\"v-spacer\\"></div>
        <div data-v-b3fa301d=\\"\\" class=\\"v-col-md-auto v-col ml-auto\\"><button data-v-b3fa301d=\\"\\" type=\\"button\\" class=\\"v-btn v-btn--elevated v-theme--light bg-primary v-btn--density-default v-btn--size-default v-btn--variant-elevated\\" data-test=\\"change-data-btn\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
            <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"> Change Data </span>
            <!---->
            <!---->
          </button>
          <!--v-if-->
        </div>
      </div>
      <div data-v-b3fa301d=\\"\\" class=\\"mt-4 pl-4 pr-4\\">
        <div data-v-b3fa301d=\\"\\" class=\\"v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--disabled v-text-field v-input--plain-underlined\\" data-test=\\"name-text\\">
          <!---->
          <div class=\\"v-input__control\\">
            <div class=\\"v-field v-field--disabled v-field--variant-underlined v-theme--light v-locale--is-ltr\\">
              <div class=\\"v-field__overlay\\"></div>
              <div class=\\"v-field__loader\\">
                <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                  <!---->
                  <div class=\\"v-progress-linear__background\\"></div>
                  <div class=\\"v-progress-linear__buffer\\" style=\\"width: 0%;\\"></div>
                  <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                    <div class=\\"v-progress-linear__indeterminate\\">
                      <div class=\\"v-progress-linear__indeterminate long\\"></div>
                      <div class=\\"v-progress-linear__indeterminate short\\"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                <!----><label class=\\"v-label v-field-label\\" for=\\"input-1\\">
                  <!---->Name
                </label>
                <!----><input disabled=\\"\\" size=\\"1\\" type=\\"text\\" id=\\"input-1\\" aria-describedby=\\"input-1-messages\\" required=\\"\\" class=\\"v-field__input\\">
                <!---->
              </div>
              <!---->
              <!---->
              <div class=\\"v-field__outline\\">
                <!----><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-1\\">
                  <!---->Name
                </label>
              </div>
            </div>
          </div>
          <!---->
          <div class=\\"v-input__details\\">
            <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-1-messages\\">
              <!---->
            </transition-group-stub>
            <!---->
          </div>
        </div>
        <div data-v-b3fa301d=\\"\\" class=\\"v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--disabled v-text-field v-input--plain-underlined\\" data-test=\\"username-text\\">
          <!---->
          <div class=\\"v-input__control\\">
            <div class=\\"v-field v-field--disabled v-field--variant-underlined v-theme--light v-locale--is-ltr\\">
              <div class=\\"v-field__overlay\\"></div>
              <div class=\\"v-field__loader\\">
                <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                  <!---->
                  <div class=\\"v-progress-linear__background\\"></div>
                  <div class=\\"v-progress-linear__buffer\\" style=\\"width: 0%;\\"></div>
                  <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                    <div class=\\"v-progress-linear__indeterminate\\">
                      <div class=\\"v-progress-linear__indeterminate long\\"></div>
                      <div class=\\"v-progress-linear__indeterminate short\\"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                <!----><label class=\\"v-label v-field-label\\" for=\\"input-3\\">
                  <!---->Username
                </label>
                <!----><input disabled=\\"\\" size=\\"1\\" type=\\"text\\" id=\\"input-3\\" aria-describedby=\\"input-3-messages\\" required=\\"\\" class=\\"v-field__input\\">
                <!---->
              </div>
              <!---->
              <!---->
              <div class=\\"v-field__outline\\">
                <!----><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-3\\">
                  <!---->Username
                </label>
              </div>
            </div>
          </div>
          <!---->
          <div class=\\"v-input__details\\">
            <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-3-messages\\">
              <!---->
            </transition-group-stub>
            <!---->
          </div>
        </div>
        <div data-v-b3fa301d=\\"\\" class=\\"v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--disabled v-text-field v-input--plain-underlined\\" data-test=\\"email-text\\">
          <!---->
          <div class=\\"v-input__control\\">
            <div class=\\"v-field v-field--disabled v-field--variant-underlined v-theme--light v-locale--is-ltr\\">
              <div class=\\"v-field__overlay\\"></div>
              <div class=\\"v-field__loader\\">
                <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                  <!---->
                  <div class=\\"v-progress-linear__background\\"></div>
                  <div class=\\"v-progress-linear__buffer\\" style=\\"width: 0%;\\"></div>
                  <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                    <div class=\\"v-progress-linear__indeterminate\\">
                      <div class=\\"v-progress-linear__indeterminate long\\"></div>
                      <div class=\\"v-progress-linear__indeterminate short\\"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                <!----><label class=\\"v-label v-field-label\\" for=\\"input-5\\">
                  <!---->Email
                </label>
                <!----><input disabled=\\"\\" size=\\"1\\" type=\\"text\\" id=\\"input-5\\" aria-describedby=\\"input-5-messages\\" required=\\"\\" class=\\"v-field__input\\">
                <!---->
              </div>
              <!---->
              <!---->
              <div class=\\"v-field__outline\\">
                <!----><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-5\\">
                  <!---->Email
                </label>
              </div>
            </div>
          </div>
          <!---->
          <div class=\\"v-input__details\\">
            <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-5-messages\\">
              <!---->
            </transition-group-stub>
            <!---->
          </div>
        </div>
        <div data-v-b3fa301d=\\"\\" class=\\"v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--disabled v-text-field v-input--plain-underlined\\" data-test=\\"recovery-email-text\\">
          <!---->
          <div class=\\"v-input__control\\">
            <div class=\\"v-field v-field--disabled v-field--variant-underlined v-theme--light v-locale--is-ltr\\">
              <div class=\\"v-field__overlay\\"></div>
              <div class=\\"v-field__loader\\">
                <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                  <!---->
                  <div class=\\"v-progress-linear__background\\"></div>
                  <div class=\\"v-progress-linear__buffer\\" style=\\"width: 0%;\\"></div>
                  <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                    <div class=\\"v-progress-linear__indeterminate\\">
                      <div class=\\"v-progress-linear__indeterminate long\\"></div>
                      <div class=\\"v-progress-linear__indeterminate short\\"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                <!----><label class=\\"v-label v-field-label\\" for=\\"input-7\\">
                  <!---->Recovery Email
                </label>
                <!----><input disabled=\\"\\" size=\\"1\\" type=\\"text\\" id=\\"input-7\\" aria-describedby=\\"input-7-messages\\" class=\\"v-field__input\\">
                <!---->
              </div>
              <!---->
              <!---->
              <div class=\\"v-field__outline\\">
                <!----><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-7\\">
                  <!---->Recovery Email
                </label>
              </div>
            </div>
          </div>
          <!---->
          <div class=\\"v-input__details\\">
            <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-7-messages\\">
              <!---->
            </transition-group-stub>
            <!---->
          </div>
        </div>
      </div>
      <hr data-v-b3fa301d=\\"\\" class=\\"v-divider v-theme--light mt-6\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
      <hr data-v-b3fa301d=\\"\\" class=\\"v-divider v-theme--light mb-6\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
      <div data-v-b3fa301d=\\"\\" class=\\"v-row\\">
        <div data-v-b3fa301d=\\"\\" class=\\"v-col\\" data-test=\\"password-header\\">
          <h3 data-v-b3fa301d=\\"\\">Password</h3>
        </div>
        <div data-v-b3fa301d=\\"\\" class=\\"v-spacer\\"></div>
        <div data-v-b3fa301d=\\"\\" class=\\"v-col-md-auto v-col ml-auto\\"><button data-v-b3fa301d=\\"\\" type=\\"button\\" class=\\"v-btn v-btn--elevated v-theme--light bg-primary v-btn--density-default v-btn--size-default v-btn--variant-elevated\\" data-test=\\"change-password-btn\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
            <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"> Change Password </span>
            <!---->
            <!---->
          </button>
          <!--v-if-->
        </div>
      </div>
      <div data-v-b3fa301d=\\"\\" class=\\"mt-4 pl-4 pr-4\\">
        <div data-v-b3fa301d=\\"\\" class=\\"v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--disabled v-text-field v-input--plain-underlined mb-4\\" data-test=\\"password-text\\">
          <!---->
          <div class=\\"v-input__control\\">
            <div class=\\"v-field v-field--disabled v-field--variant-underlined v-theme--light v-locale--is-ltr\\">
              <div class=\\"v-field__overlay\\"></div>
              <div class=\\"v-field__loader\\">
                <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                  <!---->
                  <div class=\\"v-progress-linear__background\\"></div>
                  <div class=\\"v-progress-linear__buffer\\" style=\\"width: 0%;\\"></div>
                  <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                    <div class=\\"v-progress-linear__indeterminate\\">
                      <div class=\\"v-progress-linear__indeterminate long\\"></div>
                      <div class=\\"v-progress-linear__indeterminate short\\"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                <!----><label class=\\"v-label v-field-label\\" for=\\"input-10\\">
                  <!---->Current password
                </label>
                <!----><input disabled=\\"\\" size=\\"1\\" type=\\"password\\" id=\\"input-10\\" aria-describedby=\\"input-10-messages\\" required=\\"\\" class=\\"v-field__input\\">
                <!---->
              </div>
              <!---->
              <!---->
              <div class=\\"v-field__outline\\">
                <!----><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-10\\">
                  <!---->Current password
                </label>
              </div>
            </div>
          </div>
          <div class=\\"v-input__append\\"><i class=\\"mdi-eye-off mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" tabindex=\\"0\\" aria-label=\\"Current password appended action\\"></i>
            <!---->
          </div>
          <div class=\\"v-input__details\\">
            <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-10-messages\\">
              <!---->
            </transition-group-stub>
            <!---->
          </div>
        </div>
        <div data-v-b3fa301d=\\"\\" class=\\"v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--disabled v-text-field v-input--plain-underlined mb-4\\" data-test=\\"newPassword-text\\">
          <!---->
          <div class=\\"v-input__control\\">
            <div class=\\"v-field v-field--disabled v-field--variant-underlined v-theme--light v-locale--is-ltr\\">
              <div class=\\"v-field__overlay\\"></div>
              <div class=\\"v-field__loader\\">
                <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                  <!---->
                  <div class=\\"v-progress-linear__background\\"></div>
                  <div class=\\"v-progress-linear__buffer\\" style=\\"width: 0%;\\"></div>
                  <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                    <div class=\\"v-progress-linear__indeterminate\\">
                      <div class=\\"v-progress-linear__indeterminate long\\"></div>
                      <div class=\\"v-progress-linear__indeterminate short\\"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                <!----><label class=\\"v-label v-field-label\\" for=\\"input-12\\">
                  <!---->New password
                </label>
                <!----><input disabled=\\"\\" size=\\"1\\" type=\\"password\\" id=\\"input-12\\" aria-describedby=\\"input-12-messages\\" required=\\"\\" class=\\"v-field__input\\">
                <!---->
              </div>
              <!---->
              <!---->
              <div class=\\"v-field__outline\\">
                <!----><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-12\\">
                  <!---->New password
                </label>
              </div>
            </div>
          </div>
          <div class=\\"v-input__append\\"><i class=\\"mdi-eye-off mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" tabindex=\\"0\\" aria-label=\\"New password appended action\\"></i>
            <!---->
          </div>
          <div class=\\"v-input__details\\">
            <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-12-messages\\">
              <!---->
            </transition-group-stub>
            <!---->
          </div>
        </div>
        <div data-v-b3fa301d=\\"\\" class=\\"v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--disabled v-text-field v-input--plain-underlined mb-4\\" data-test=\\"confirmNewPassword-text\\">
          <!---->
          <div class=\\"v-input__control\\">
            <div class=\\"v-field v-field--disabled v-field--variant-underlined v-theme--light v-locale--is-ltr\\">
              <div class=\\"v-field__overlay\\"></div>
              <div class=\\"v-field__loader\\">
                <div class=\\"v-progress-linear v-theme--light v-locale--is-ltr\\" style=\\"top: 0px; height: 0px; --v-progress-linear-height: 2px;\\" role=\\"progressbar\\" aria-hidden=\\"true\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\">
                  <!---->
                  <div class=\\"v-progress-linear__background\\"></div>
                  <div class=\\"v-progress-linear__buffer\\" style=\\"width: 0%;\\"></div>
                  <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                    <div class=\\"v-progress-linear__indeterminate\\">
                      <div class=\\"v-progress-linear__indeterminate long\\"></div>
                      <div class=\\"v-progress-linear__indeterminate short\\"></div>
                    </div>
                  </transition-stub>
                  <!---->
                </div>
              </div>
              <!---->
              <div class=\\"v-field__field\\" data-no-activator=\\"\\">
                <!----><label class=\\"v-label v-field-label\\" for=\\"input-14\\">
                  <!---->Confirm new password
                </label>
                <!----><input disabled=\\"\\" size=\\"1\\" type=\\"password\\" id=\\"input-14\\" aria-describedby=\\"input-14-messages\\" required=\\"\\" class=\\"v-field__input\\">
                <!---->
              </div>
              <!---->
              <!---->
              <div class=\\"v-field__outline\\">
                <!----><label class=\\"v-label v-field-label v-field-label--floating\\" aria-hidden=\\"true\\" for=\\"input-14\\">
                  <!---->Confirm new password
                </label>
              </div>
            </div>
          </div>
          <div class=\\"v-input__append\\"><i class=\\"mdi-eye-off mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable\\" role=\\"button\\" aria-hidden=\\"false\\" tabindex=\\"0\\" aria-label=\\"Confirm new password appended action\\"></i>
            <!---->
          </div>
          <div class=\\"v-input__details\\">
            <transition-group-stub name=\\"slide-y-transition\\" tag=\\"div\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" class=\\"v-messages\\" role=\\"alert\\" aria-live=\\"polite\\" id=\\"input-14-messages\\">
              <!---->
            </transition-group-stub>
            <!---->
          </div>
        </div>
      </div>
      <hr data-v-b3fa301d=\\"\\" class=\\"v-divider v-theme--light mt-6\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
      <hr data-v-b3fa301d=\\"\\" class=\\"v-divider v-theme--light mb-6\\" aria-orientation=\\"horizontal\\" role=\\"separator\\">
      <!--v-if-->
    </div>
  </div>
</div>"
`;
